spec/unit/rest/Twilio.spec.js

```diff
@@ -1,87 +1,137 @@
+          .then(() => scope.done());
+      });
+      it('should set the region and edge properly', () => {
+        const scope = nock('https://api.edge.region.twilio.com')
+          .get('/')
+          .reply(200, 'test response');
+        client.edge = 'edge';
+        client.region = 'region';
+        return client.request({method: 'GET', uri: 'https://api.twilio.com'})
+          .then(() => scope.done());
+      });
+      it('should set the region and edge properly when an edge is already included', () => {
+        const scope = nock('https://api.edge2.region.twilio.com')
+          .get('/')
+          .reply(200, 'test response');
+        client.edge = 'edge2';
+        return client.request({method: 'GET', uri: 'https://api.edge1.region.twilio.com'})
+          .then(() => scope.done());
+      });
+      it('should set the region and edge properly when a region is already included', () => {
+        const scope = nock('https://api.edge.region2.twilio.com')
+          .get('/')
+          .reply(200, 'test response');
+        client.region = 'region2';
+        return client.request({method: 'GET', uri: 'https://api.edge.region.twilio.com'})
+          .then(() => scope.done());
+      });
+      it('should set the region properly when a region is already included', () => {
+        const scope = nock('https://api.region2.twilio.com')
+          .get('/')
+          .reply(200, 'test response');
+        client.region = 'region2';
+        return client.request({method: 'GET', uri: 'https://api.region.twilio.com'})
+          .then(() => scope.done());
+      });
+      it('should set the region properly on a custom domain', () => {
+        const scope = nock('https://api.region2.domain.com')
+          .get('/')
+          .reply(200, 'test response');
+        client.region = 'region2';
+        return client.request({method: 'GET', uri: 'https://api.domain.com'})
+          .then(() => scope.done());
+      });
+      it('should set the region properly when a port is included', () => {
+        const scope = nock('https://api.region.twilio.com:123')
+          .get('/')
+          .reply(200, 'test response');
+        client.region = 'region';
+        return client.request({method: 'GET', uri: 'https://api.twilio.com:123'})
+          .then(() => scope.done());
+      });
     });
-    it('should set the region properly if only the region is specified', () => {
-      const scope = nock('https://api.region.twilio.com')
+  });
+
+  describe('adding user agent extensions', () => {
+    const defaultUserAgent = util.format(
+      'twilio-node/%s (%s %s) node/%s',
+      moduleInfo.version,
+      os.platform(),
+      os.arch(),
+      process.version
+    );
+
+    it('sets the user-agent by default', () => {
+      const client = new twilio('ACXXXXXXXX', 'test-password');
+      const scope = nock('https://api.twilio.com', {
+        reqheaders: {
+          'User-Agent': defaultUserAgent,
```

Comment:
Can we test what this value should be in this case, even if it's just a pattern match?

Reply:
Fair point, using the exact same code to test the user agent as that which generates it isn't cool.

How about testing it by building a string in a different way to show the expectation a bit clearer?